src/bidiMapper/domains/network/NetworkManager.ts

```diff
@@ -22,129 +22,118 @@
  */
 import type Protocol from 'devtools-protocol';
 
-import type {ICdpClient} from '../../../cdp/cdpClient.js';
-import type {EventManager} from '../events/EventManager.js';
-import {DefaultMap} from '../../../utils/DefaultMap.js';
 import type {Network} from '../../../protocol/protocol.js';
+import type {CdpTarget} from '../context/cdpTarget.js';
 
-import {NetworkRequest} from './NetworkRequest.js';
+import type {NetworkRequest} from './NetworkRequest.js';
+import type {NetworkStorage} from './NetworkStorage.js';
 
+/** Maps 1:1 to CdpTarget. */
 export class NetworkManager {
-  readonly #eventManager: EventManager;
+  readonly #cdpTarget: CdpTarget;
+  readonly #networkStorage: NetworkStorage;
 
-  /**
-   * Map of request ID to NetworkRequest objects. Needed as long as information
-   * about requests comes from different events.
-   */
-  readonly #requestMap: DefaultMap<Network.Request, NetworkRequest>;
-
-  private constructor(eventManager: EventManager) {
-    this.#eventManager = eventManager;
+  private constructor(cdpTarget: CdpTarget, networkStorage: NetworkStorage) {
+    this.#cdpTarget = cdpTarget;
+    this.#networkStorage = networkStorage;
+  }
 
-    this.#requestMap = new DefaultMap(
-      (requestId) => new NetworkRequest(requestId, this.#eventManager)
-    );
+  /** Returns the CDP Target associated with this NetworkManager instance. */
+  get cdpTarget(): CdpTarget {
+    return this.#cdpTarget;
   }
 
   static create(
-    cdpClient: ICdpClient,
-    eventManager: EventManager
+    cdpTarget: CdpTarget,
+    networkStorage: NetworkStorage
   ): NetworkManager {
-    const networkProcessor = new NetworkManager(eventManager);
+    const networkManager = new NetworkManager(cdpTarget, networkStorage);
 
-    cdpClient
+    cdpTarget.cdpClient
       .browserClient()
       .on(
         'Target.detachedFromTarget',
         (params: Protocol.Target.DetachedFromTargetEvent) => {
-          if (cdpClient.sessionId === params.sessionId) {
-            networkProcessor.dispose();
+          if (cdpTarget.cdpClient.sessionId === params.sessionId) {
+            networkManager.#networkStorage.disposeRequestMap();
```

Comment:
This I am not sure as from my understanding it will clear all requests. This will work if a single page is open but what if we open 2 pages? Happy to merge if the test pass but lets add comment or issue to investigate further.

Reply:
This diff is a no-op. I just switched to cdpTarget.cdpClient instead of cdpClient; and to use the method within network storage, otherwise nothing else changed.

I can add a comment anyway.